lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -924,6 +924,7 @@ export class NormalComponent<
 
     const pcb_component = db.pcb_component.get(this.pcb_component_id!)
 
+    if (cadModel === "") return
```

Comment:
Dont use empty strings

```

Avoid using empty strings ("") to represent absence, nullability, or special states. Use null, undefined, or omit the field instead.
```

```suggestion
    if (cadModel === null) return
```